hlapipeline/alignimages.py

```diff
@@ -3,65 +3,181 @@
+    # for alignment purposes.
+    if filteredTable['doProcess'].sum() == 0:
+        print("No viable images in filtered table - no processing done.\n")
+        return
+
+    # Get the list of all "good" files to use for the alignment
+    processList = filteredTable['imageName'][np.where(filteredTable['doProcess'])]
+    processList = list(processList) #Convert processList from numpy list to regular python list
+    print("\nSUCCESS")
 
     # 3: Build WCS for full set of input observations
-    refwcs = amutils.build_reference_wcs(imglist)
+    print("-------------------- STEP 3: Build WCS --------------------")
+    refwcs = amutils.build_reference_wcs(processList)
+    print("\nSUCCESS")
 
     # 4: Retrieve list of astrometric sources from database
-    reference_catalog = generate_astrometric_catalog(imglist, catalog='GAIADR2')
-
+    print("-------------------- STEP 4: Detect astrometric sources --------------------")
+    # While loop to accommodate using multiple catalogs
+    doneFitting = False
+    catalogIndex = 0
+    while not doneFitting:
+        reference_catalog = generate_astrometric_catalog(processList, catalog=catalogList[catalogIndex])
+
+        # The table must have at least MIN_CATALOG_THRESHOLD entries to be useful
+        if len(reference_catalog) < MIN_CATALOG_THRESHOLD:
+            print("Not enough sources found in catalog" + catalogList[catalogIndex])
+            catalogIndex += 1
+            if catalogIndex <= (numCatalogs - 1):
+                print("Try again with the next catalog")
+                continue
+            else:
+                print("Not enough sources found in any catalog - no processing done.")
+                return
+        else:
+            doneFitting = True
+    print("\nSUCCESS")
     # 5: Extract catalog of observable sources from each input image
-    extracted_sources = generate_source_catalogs(imglist, refwcs, threshold=1000)
+    print("-------------------- STEP 5: Source finding --------------------")
+    extracted_sources = generate_source_catalogs(processList)
+    for imgname in extracted_sources.keys():
+        # The catalog of observable sources must have at least MIN_OBSERVABLE_THRESHOLD entries to be useful
+        # TODO *** Is this no hope or iterate on threshold or ???
+        if len(extracted_sources[imgname]["catalog_table"]) < MIN_OBSERVABLE_THRESHOLD:
+            print("Not enough sources ({}) found in image {}".format(len(extracted_sources[imgname]["catalog_table"]),imgname))
+            return
+
+    print("\nSUCCESS")
 
     # 6: Cross-match source catalog with astrometric reference source catalog
+    # TODO *** catalog cross match call here
+    # if len(out_catalog) < MIN_CROSS_MATCHS:
+    #     print("Not enough cross matches found between astrometric catalog and sources found in images")
+    #     print("Try again with the next catalog")
+    #     catalogIndex += 1
+    #     continue
 
```

Comment:
oops.  Need to make sure there is another catalog to use.  Similar catalog checking here as done at Lines 150-154.

Reply:
I am going to pull Michael's branch so that I can better see what is going on and attempt to flesh out the looping logic.

Reply:
issue resolved in commit [d8ad5cd](https://github.com/spacetelescope/hlapipeline/pull/36/commits/d8ad5cd11b05df27d4ff7e5580593807a7485309).